third_party/cabal2bazel/bzl/cabal_package.bzl

```diff
@@ -15,124 +15,58 @@
-    "directory",
-    "filepath",
-    "ghc",
-    "ghc-boot",
-    "ghc-boot-th",
-    "ghc-prim",
-    "ghci",
-    "haskeline",
-    "hoopl",
-    "hpc",
-    "integer-gmp",
-    "pretty",
-    "process",
-    "template-haskell",
-    "terminfo",
-    "time",
-    "transformers",
-    "unix",
-]
+load("@bazel_skylib//:lib.bzl", "paths")
+load("@io_tweag_rules_haskell//haskell:haskell.bzl",
+     "haskell_library",
+     "haskell_binary")
+load(":bzl/alex.bzl", "genalex")
+load(":bzl/cabal_paths.bzl", "cabal_paths")
+load(":bzl/happy.bzl", "genhappy")
 
 _conditions_default = "//conditions:default"
 
-# A list of default packages which are hidden by default.
-_hidden_default_packages = ["ghc"]
-
-# Extra GHC arguments to undo the google3 defaults which might otherwise break
-# some Cabal packages.
-_undo_default_options = [
-    "-XForeignFunctionInterface",
-    "-XNoScopedTypeVariables",
-]
-_ignored_extensions = [
-    # Deprecated in July 2014: https://ghc.haskell.org/trac/ghc/ticket/9242#comment:28
-    "OverlappingInstances",
-]
-
-# cabal_haskell_library is a build rule which is a thin wrapper around
-# haskell_library, tagging it with a version number.
-# Usage:
-#   haskell_library(
-#       name = "foo-lib",
-#       ...,
-#   )
-#   cabal_haskell_library(
-#       name = "foo",
-#       version = "1.2.3",
-#   )
-#   # Forward a library, in particular used for version-agnostic libraries.
-#   # Automaticaly picks up the version number from the "library" attribute.
-#   cabal_haskell_library(
-#      library = ":foo-forwarded",
-#   )
-def _impl_cabal_haskell_library(ctx):
-  if ctx.attr.version:
-    cabal_id = struct(
-        name=ctx.label.name,
-        version=ctx.attr.version)
-  else:
-    cabal_id = ctx.attr.library.haskell_cabal_id
-
-  return forward_deps(
-      ctx,
-      ctx.attr.deps,
-      {"haskell_cabal_id": cabal_id})
-
-_cabal_haskell_library = rule(
-    implementation=_impl_cabal_haskell_library,
-    attrs={
-        "deps": attr.label_list(),
-        "library": attr.label(),
-        "version": attr.string(),
-    },
-)
-
-def cabal_haskell_library(name, library, version=""):
-  # We want cabal_haskell_library to accept only one dependency, the library
-  # that it is forwarding, so we use a single label attr called "library"
-  # instead of "deps". However, bazel treats "deps" specially when finding
-  # transitive data dependencies and if we don't include the "library"
-  # dependency in a "deps" attribute, we must manually forward all data
-  # dependencies. To avoid that, we include a "deps" attr and use this macro to
-  # enforce that "deps == [library]".
-  _cabal_haskell_library(
-      name=name,
-      deps=[library],
-      library=library,
-      version=version,
-  )
-
 # Cabal macro generation target name ends with this.
 _macros_suffix = "-macros"
 
+# Template files that we should install manually for Happy and Alex
+# TODO: figure out a better bootstrapping method
+_MANUAL_DATA_FILES = {
```

Comment:
Does it make sense to host these lists in '@haskell_{happy,alex}//' instead of coupling things up tightly?

Reply:
I moved the list into `templates/templates.bzl`.  Right now `@haskell_*` are auto-generated so the source of truth needs to live somewhere in this repo.